### DISCUSSION TITLE
Trace/Span in JSON Logs not capturing in open telemetry collector

### DISCUSSION THREAD

**Discussion by kamaldkannan** [open-telemetry/opentelemetry-collector#5133] [General]
Please help to advise why Trace/Span in JSON Logs are not capturing in open telemetry collector

Log Published
{"@timestamp":"2022-03-31T18:55:01.290+00:00","severity":"INFO","service":"test","traceId":"f61e7a3e224cc5dc","spanId":"70aa4d30d3b3dba9","pid":"27060","thread":"thread1","class":"c.c.c.t.p.k.f.FunctionConfiguration","messsage":"valid message received"}

Collector Log
Resource SchemaURL: https://opentelemetry.io/schemas/1.8.0
Resource labels:
     -> host.arch: STRING(amd64)
     -> host.name: ***
     -> os.description: ****
     -> os.type: STRING(windows)
     -> process.command_line: ******
     -> process.executable.path: ***
     -> process.pid: ***
     -> process.runtime.description: ***
     -> process.runtime.name: STRING(Java(TM) SE Runtime Environment)
     -> process.runtime.version: STRING(11.0.10+8-LTS-162)
     -> service.appid: STRING(175828)
     -> service.name: ***
     -> telemetry.auto.version: STRING(1.11.0)
     -> telemetry.sdk.language: STRING(java)
     -> telemetry.sdk.name: STRING(opentelemetry)
     -> telemetry.sdk.version: STRING(1.11.0)
InstrumentationLibraryLogs #0
InstrumentationLibraryLogs SchemaURL: 
InstrumentationLibrary ****
LogRecord #0
Timestamp: 2022-03-31 18:55:01.29 +0000 UTC
Severity: INFO
ShortName: 
Body: valid message received
**Trace ID: 
Span ID:** 
Flags: 0

Collector config

otel-collector-config: |
    receivers:
      otlp:
        protocols:
          grpc:
          http:
    processors:
      batch:
      memory_limiter:
        # 80% of maximum memory up to 2G
        limit_mib: 1500
        # 25% of limit up to 2G
        spike_limit_mib: 512
        check_interval: 5s
    exporters:

     logging:
        loglevel: debug
        sampling_initial: 5
        sampling_thereafter: 200

  logs/1:
          receivers: [otlp]
          processors: [batch]
          exporters: [logging]

**Comment by pmm-sumo** [comment]
I think this might something related to instrumentation not including the Span ID / Trace ID. I just did a quick test with opentelemetry-java using log emitter directly, i.e.

```
Span exampleSpan = tracer.spanBuilder("exampleSpan").startSpan();
try (Scope scope = exampleSpan.makeCurrent()) { 
 
 logProvider.logEmitterBuilder("").build().logBuilder().setBody("foo").setContext(Context.current()).emit();
}
```

And the output in the collector was: 
```
Body: foo
Trace ID: ce64911bae1ebd5c18971e8842ba5974
Span ID: 2b1e9901b17123e2
```

Could you provide more details on the instrumentation setup? Also, I think this question might be more fit for [opentelemetry-java](https://github.com/open-telemetry/opentelemetry-java/discussions)
